docs/technical-design/howto-migrations.md

```diff
@@ -0,0 +1,48 @@
+- Run in all environments, in as close to a way as they run in CI.
+- Write migrations that can be tested repeatedly.
+- Avoid making migrations directly dependent on a specific feature in progress being available in production.
+
+## Data Migrations
+Currently data migrations change existing records in the database. They are run as standalone lambdas that developers must manually trigger by environment.
+
+### Steps
+1. Prepare for manual testing in lower environments.
+    - Build a PR review app off `main` to start out. Try to populate with submissions similar to what is in production (For example, populate submissions via Cypress and there are specific submissions types you know you will need to test, make sure add them).
+1. Write the migration with verbose logs.
+    - The migration will be written as a [Node lambda](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html) and live in `app-api/src/handlers`.  Make sure you use ES6 async await.
+    - Console statements are essential because there are line numbers in Cloudwatch when the migration fails or does not apply correctly. There is little additional context about why a run execution paused or failed.
+    - Here is an example of verbose debug console statements wrapping a loop:
+    ![debug consoles in data migration script](../../.images/verbose-logs-example.png)
+1. Write unit tests.
+    - Consider including a test for `migration can be run repeatedly without data loss or unexpected results". Using repeatable migrations (when possible) makes it easier on developers and easier to unwind work.
+1. Manually test the migration in your review app.
+    - Log into [AWS Lambda Console](https://console.aws.amazon.com/lambda/home) and find the lambda. This means choosing `app-api-` lambda with your branch name and migration name included.
+    - Click the Test tab and the `Test` button. Use the generic hello world event.
+        - All output will appear inline on the same page as the lambda run. All consoles appear there as well. You can also click into a link from there into Cloudwatch to be able to see entire output of logs around the lambda execution.
+     - If you need to debug quickly, without waiting for redeploy in review app use the severless CLI script.
+        -  `serverless deploy function --function name_of_migration --stage my_branch && echo "DEPLOYED" && serverless logs --function migrate_rate_documents  --stage my_branch --tail` can b e run in your `app-api` directory. This script runs serverless off your local API code (configured with your env variables) with the AWS resources associated with your branch.  `my_branch`is the name of your deployed Github branch and `name_of_migration` is the name of the migration file mind the file path.
+        - If this is to be run in DEV (`main`) you will have to reshape your `envrc.local` to imitate configuration on that stage.
```

Comment:
This might not make a lot of sense to someone who doesn't already know what it means. Maybe include a screenshot of a lambda config?  Let me see if this paste works.
<img width="1598" alt="image" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/8964335/befc9607-c3d2-4c02-b75c-eef0a5e8f2db">